src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -401,37 +403,84 @@ private static List<AnnotationNode> createClassAnnotations(
+            .setValueExpr(
+                ValueExpr.withValue(
+                    PrimitiveValue.builder().setValue("false").setType(TypeNode.BOOLEAN).build()))
+            .build();
     AnnotationNode conditionalOnPropertyNode =
         AnnotationNode.builder()
             .setType(types.get("ConditionalOnProperty"))
-            .setDescription(
-                "value = \""
-                    + Utils.springPropertyPrefix(libName, service.name())
-                    + ".enabled\", matchIfMissing = false")
+            .addDescription(valueStringAssignmentExpr)
+            .addDescription(matchIfMissingAssignmentExpr)
+            .build();
+
+    TypeNode clazzType =
+        TypeNode.withReference(
+            VaporReference.builder()
+                .setName(ClassNames.getServiceClientClassName(service))
+                .setPakkage(service.pakkage())
+                .build());
+    VariableExpr conditionalOnClassVariableExpr =
+        VariableExpr.builder()
+            .setVariable(Variable.builder().setType(TypeNode.CLASS_OBJECT).setName("class").build())
+            .setStaticReferenceType(clazzType)
             .build();
     AnnotationNode conditionalOnClassNode =
         AnnotationNode.builder()
             .setType(types.get("ConditionalOnClass"))
-            .setDescription(
-                "value = "
-                    + ClassNames.getServiceClientClassName(service)
-                    + ".class") // TODO: change after annotation feature merged. need to produce
-            // XXX.class
+            .setDescription(conditionalOnClassVariableExpr)
+            .build();
+
+    AssignmentExpr proxyBeanMethodsAssignmentExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(
+                VariableExpr.withVariable(
+                    Variable.builder()
+                        .setName("proxyBeanMethods")
+                        .setType(TypeNode.BOOLEAN)
+                        .build()))
+            .setValueExpr(
+                ValueExpr.withValue(
+                    PrimitiveValue.builder().setValue("false").setType(TypeNode.BOOLEAN).build()))
             .build();
     AnnotationNode configurationNode =
         AnnotationNode.builder()
             .setType(types.get("Configuration"))
-            .setDescription("proxyBeanMethods = false") // TODO: change to parameters
+            .addDescription(proxyBeanMethodsAssignmentExpr)
+            .build();
+
+    TypeNode propertiesClazzType =
+        TypeNode.withReference(
+            VaporReference.builder()
+                .setName(types.get(service.name() + "Properties").reference().name())
+                .setPakkage(service.pakkage())
+                .build());
+    VariableExpr propertiesClassVariableExpr =
+        VariableExpr.builder()
+            .setVariable(Variable.builder().setType(TypeNode.CLASS_OBJECT).setName("class").build())
+            .setStaticReferenceType(propertiesClazzType)
```

Comment:
I have types are defined in 2 ways at the bottom of class:
- [dynamic types](https://github.com/googleapis/gapic-generator-java/blob/62cd6384e2afa58ee7b2f8d6c452f15e0ac05223/src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java#L984) - these are types that needs to use vapor references.
- [static types](https://github.com/googleapis/gapic-generator-java/blob/62cd6384e2afa58ee7b2f8d6c452f15e0ac05223/src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java#L966) - these dependencies are brought in to gapic-generator so the concrete types can be used.  One caveat is that I am also working on #1046 to bring in Spring dependencies so those can be static instead of dynamic.

In this case, the properties class is already defined in [here](https://github.com/googleapis/gapic-generator-java/blob/62cd6384e2afa58ee7b2f8d6c452f15e0ac05223/src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java#L1002-L1007), so you can use `types.get(service.name() + "Properties")` here and get rid of `propertiesClazzType`.

Sorry about the confusion that `types.get(service.name() + "Properties").reference().name()` brings, I was trying to remind myself this type is avalaible.

Reply:
Thanks for the explanation here - I hadn't noticed the typeMaps until recently, will go ahead and make this update! 

Will keep an eye on #1046 and maybe rebase this branch if you wanted to merge that first though, since these two PRs will probably have conflicting changes.